src/transforms/kubernetes/mod.rs

```diff
@@ -1 +1,568 @@
+            .collect()
+    }
+
+    fn with_prefix(name: &str) -> String {
+        event::log_schema().kubernetes_key().as_ref().to_owned() + "." + name
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::Field;
+
+    #[test]
+    fn field_name() {
+        assert_eq!("priority_class_name", &Field::PriorityClassName.name());
+    }
+}
+
+#[cfg(test)]
+mod integration_tests {
+    #![cfg(feature = "kubernetes-integration-tests")]
+
+    use crate::sources::kubernetes::test::{echo, logs, user_namespace, Kube, VECTOR_YAML};
+    use crate::test_util::{random_string, wait_for};
+    use kube::api::{Api, RawApi};
+    use uuid::Uuid;
+
+    static NAME_MARKER: &'static str = "$(NAME)";
+    static FIELD_MARKER: &'static str = "$(FIELD)";
+
+    static ROLE_BINDING_YAML: &'static str = r#"
+# Permissions to use Kubernetes API.
+# Necessary for kubernetes_pod_metadata transform.
+# Requires that RBAC authorization is enabled.
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: $(NAME)
+subjects:
+- kind: ServiceAccount
+  name: default
+  namespace: $(TEST_NAMESPACE)
+roleRef:
+  kind: ClusterRole
+  name: view
+  apiGroup: rbac.authorization.k8s.io
+"#;
+
+    static CONFIG_MAP_YAML_WITH_METADATA: &'static str = r#"
+# ConfigMap which contains vector.toml configuration for pods.
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: vector-config
+  namespace: $(TEST_NAMESPACE)
+data:
+  vector-agent-config: |
+    # VECTOR.TOML
+    # Configuration for vector-agent
+
+    # Set global options
+    data_dir = "/tmp/vector/"
+
+    # Ingest logs from Kubernetes
+    [sources.kubernetes_logs]
+      type = "kubernetes"
+
+    [transforms.kube_metadata]
+      type = "kubernetes_pod_metadata"
+      inputs = ["kubernetes_logs"]
+      $(FIELD)
+
+    [sinks.out]
+      type = "console"
+      inputs = ["kube_metadata"]
+      target = "stdout"
+
+      encoding = "json"
+      healthcheck = true
+
+  # This line is not in VECTOR.TOML
+"#;
+
+    fn cluster_role_binding_api() -> RawApi {
+        RawApi {
+            group: "rbac.authorization.k8s.io".into(),
+            resource: "clusterrolebindings".into(),
+            prefix: "apis".into(),
+            version: "v1".into(),
+            ..Default::default()
+        }
+    }
+
+    fn binding_name(namespace: &str) -> String {
+        "binding-".to_owned() + namespace
+    }
+
+    fn metadata_config_map(fields: Option<Vec<&str>>) -> String {
+        let replace = if let Some(fields) = fields {
+            format!(
+                "fields = [{}]",
+                fields
+                    .iter()
+                    .map(|field| format!("{:?}", field))
+                    .collect::<Vec<_>>()
+                    .join(",")
+            )
+        } else {
+            "".to_owned()
+        };
+
+        CONFIG_MAP_YAML_WITH_METADATA.replace(FIELD_MARKER, replace.as_str())
+    }
+
+    #[test]
+    fn kube_metadata() {
```

Comment:
This seems like quite a small test, would it make sense to add some more coverage?

Reply:
Well, this test covers all the main paths except those of failure and deletion. Now deletion could theoretically be tested, but all the ways that I see are quite complex and definitely not worth it. 

There is also a separate test for `WatchClient`.

I think those cover it nicely.